astropy_helpers/commands/build_sphinx.py

```diff
@@ -170,18 +172,22 @@ def run(self):
         # prevents a continuous updating at the terminal, but there's no
         # apparent way around this.
         if self.warnings_returncode:
-            proc = subprocess.Popen([sys.executable],
+            proc = subprocess.Popen([sys.executable, '-c', subproccode.encode('utf-8')],
                                     stdin=subprocess.PIPE,
                                     stdout=subprocess.PIPE,
                                     stderr=subprocess.STDOUT)
-            stdo, _ = proc.communicate(subproccode.encode('utf-8'))
-            stdo = stdo.decode('utf-8')
 
-            print(stdo)
+            stdolines = []
+            with proc.stdout:
+                for line in iter(proc.stdout.readline, b''):
+                    line = line.strip(b'\n')
+                    stdolines.append(line)
+                    print(line.decode('utf-8'))
 
-            stdolines = stdo.splitlines()
+            # Poll to set proc.retcode
+            proc.poll()
 
-            if 'build succeeded.' in stdolines:
+            if b'build succeeded.' in stdolines:
```

Comment:
One thing we could do to improve this, is it seems there's no reason to save the stdout lines once they've been printed.  Instead set `retcode = 1` by default, and do the check for "build succeeded" in the loop above (and if found, set `retcode = 0`.

Seems fine otherwise--thanks for fixing this.